mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/factories/FlinkCatalogFactory.java

```diff
@@ -70,31 +80,58 @@ public Catalog createCatalog(Context context) {
     final String defaultDatabase = helper.getOptions().get(DEFAULT_DATABASE);
     String metastoreUrl = helper.getOptions().get(ArcticCatalogFactoryOptions.METASTORE_URL);
 
+    String amoroCatalogName =
+        ArcticThriftUrl.parse(metastoreUrl, THRIFT_TABLE_SERVICE_NAME).catalogName();
+    UnifiedCatalog unifiedCatalog =
+        UnifiedCatalogLoader.loadUnifiedCatalog(metastoreUrl, amoroCatalogName, Maps.newHashMap());
+    Configuration hadoopConf = unifiedCatalog.authenticationContext().getConfiguration();
+
+    TableFormat catalogTableFormat =
+        TableFormat.valueOf(unifiedCatalog.properties().get(TABLE_FORMATS));
+
     Map<TableFormat, AbstractCatalog> availableCatalogs = Maps.newHashMap();
     SUPPORTED_FORMATS.forEach(
         tableFormat -> {
           if (!availableCatalogs.containsKey(tableFormat)) {
-            availableCatalogs.put(tableFormat, createCatalog(context, tableFormat));
+            if (catalogTableFormat == TableFormat.ICEBERG
+                && (tableFormat.in(TableFormat.MIXED_HIVE, TableFormat.MIXED_ICEBERG))) {
+              // Mixed catalog couldn't load the iceberg table, so specify the table formats to the
+              // mixed catalog
+              context.getOptions().put(FLINK_TABLE_FORMATS.key(), tableFormat.toString());
+            }
+            availableCatalogs.put(tableFormat, createCatalog(context, tableFormat, hadoopConf));
           }
```

Comment:
It's not necessary to initialize all supported formats in the catalog, as it's possible that not all these formats will actually be used. However, this would require corresponding dependencies in the runtime environment.